static/js/copy_and_paste.js

```diff
@@ -142,6 +142,16 @@ function copy_handler() {
 
 $(function () {
     $(document).on('copy', copy_handler);
+    $("#new_message_content").bind("paste", function(event){
+        event.preventDefault();
+        var clipboardData = (event.originalEvent || event).clipboardData;
+        var text = clipboardData.getData('text/html') || clipboardData.getData('text/plain') || document.execCommand('insertHTML', false, '<pre>' + text + '</pre>');
+        var html = toMarkdown(text);
+        var div = document.createElement("div");
+        div.innerHTML = html;
+        var text = div.textContent || div.innerText || "";
+        this.value = text;
+    });
```

Comment:
Can you add some comments explaining why this logic is correct?  There's a lot of `||` gates between different things, where it feels unlikely that all 3 versions should do the same thing.  

Also, can you add some unit tests to the `node` test suite in `frontend_tests/node_tests`?  To do this, I'd probably extract this code to a function called `exports.paste_handler` so that you can just pass in mocked-up `event` objects rather than needing to trigger `paste` events.